source/common/json/json_internal.h

```diff
@@ -38,6 +38,13 @@ class Factory {
    * @return A string suitable for inclusion in a JSON stream, including double-quotes.
    */
   static std::string serialize(absl::string_view str);
+
+  /*
+   * Serializes a JSON string to a byte vector using the MessagePack serialization format.
+   * If the provided JSON string is invalid, an empty vector will be returned.
+   * See: https://github.com/msgpack/msgpack/blob/master/spec.md
+   */
+  static std::vector<uint8_t> jsonToMsgpack(const std::string& json);
```

Comment:
I think this may throw an exception if the `json` isn't a legal json string. Please add a comment to tell it.

And I prefer to just place this utility function in the source dir of fluentd logger because it only be used by that logger.

Reply:
It will not throw an example as I'm using an exception safe method, it will return an empty vector an this is indicated in line 44: ``If the provided JSON string is invalid, an empty vector will be returned.``. Also added unit tests to verify this.